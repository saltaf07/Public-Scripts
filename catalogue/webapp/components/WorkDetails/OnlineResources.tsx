import { useEffect, useState, useRef, FunctionComponent } from 'react';
import { font, classNames } from '@weco/common/utils/classnames';
import { getItemsByLocationType } from '@weco/common/utils/works';
import WorkDetailsSection from '../WorkDetailsSection/WorkDetailsSection';
import { DigitalLocation, Work, Item } from '@weco/common/model/catalogue';
import styled from 'styled-components';

const ShowHideButton = styled.button.attrs({
  className: classNames({
    'plain-button no-margin no-padding': true,
    [font('hnl', 5)]: true,
  }),
})`
  text-decoration: underline;

  &:hover {
    text-decoration: none;
  }
`;
type Props = {
  work: Work;
};
const OnlineResources: FunctionComponent<Props> = ({ work }: Props) => {
  type ItemWithDigitalLocation = Omit<Item, 'locations'> & {
    location: DigitalLocation;
  };

  const onlineResources = getItemsByLocationType(work, 'online-resource').map(
    i => {
      const [location] = i.locations;
      return {
        title: i.title,
        location,
      };
    }
  ) as ItemWithDigitalLocation[]; // FIXME: I think this casting might be cheating?

  const firstThreeOnlineResources = onlineResources.slice(0, 3);
  const remainingOnlineResources = onlineResources.slice(3);
  const [
    isShowingRemainingOnlineResources,
    setIsShowingRemainingOnlineResources,
  ] = useState(false);
  const firstOfRemainingOnlineResourcesRef = useRef<HTMLAnchorElement>(null);
  const showHideResourcesRef = useRef<HTMLButtonElement>(null);

  useEffect(() => {
    if (isShowingRemainingOnlineResources) {
      firstOfRemainingOnlineResourcesRef?.current?.focus();
    } else {
      showHideResourcesRef?.current?.focus();
    }
  }, [isShowingRemainingOnlineResources]);

  return onlineResources.length > 0 ? (
    <WorkDetailsSection headingText="Available online">
      <ul
        className={classNames({
          'plain-list no-margin no-padding': true,
        })}
      >
        {firstThreeOnlineResources.map(item => (
          <li
            className={classNames({
              [font('hnl', 5)]: true,
            })}
            key={item.location.url}
          >
            <a href={item.location.url}>
              {item.title
                ? `${item.title}: View resource`
                : item.location.linkText}
            </a>
          </li>
        ))}
        {isShowingRemainingOnlineResources && (
          <>
            {remainingOnlineResources.map((item, index) => (
              <li
                className={classNames({
                  [font('hnl', 5)]: true,
                })}
                key={item.location.url}
              >
                <a
                  href={item.location.url}
                  ref={
                    index === 0 ? firstOfRemainingOnlineResourcesRef : undefined
                  }
                >
                  {item.title
                    ? `${item.title}: View resource`
                    : item.location.linkText}
                </a>
              </li>
            ))}
          </>
        )}
      </ul>
      {remainingOnlineResources.length > 0 && (
        <ShowHideButton
          ref={showHideResourcesRef}
          onClick={() =>
            setIsShowingRemainingOnlineResources(
              !isShowingRemainingOnlineResources
            )
          }
        >
          {isShowingRemainingOnlineResources
            ? '- Show less'
            : `+ ${remainingOnlineResources.length} more`}
        </ShowHideButton>
      )}
    </WorkDetailsSection>
  ) : null;
};

export default OnlineResources;
