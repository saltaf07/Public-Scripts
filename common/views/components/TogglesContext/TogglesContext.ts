import { createContext } from 'react';
import { Toggles } from '@weco/toggles';

const TogglesContext = createContext<Toggles>({} as Toggles);

export default TogglesContext;
