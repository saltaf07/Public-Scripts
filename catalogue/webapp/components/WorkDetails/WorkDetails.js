// @flow
import moment from 'moment';
import { type IIIFManifest } from '@weco/common/model/iiif';
import { type Work } from '@weco/common/model/work';
import { font, grid, classNames } from '@weco/common/utils/classnames';
import { downloadUrl } from '@weco/common/services/catalogue/urls';
import { worksLink } from '@weco/common/services/catalogue/routes';
import {
  getDownloadOptionsFromImageUrl,
  getDigitalLocationOfType,
  getWorkIdentifiersWith,
  getEncoreLink,
} from '@weco/common/utils/works';
import getAugmentedLicenseInfo from '@weco/common/utils/licenses';
import {
  getDownloadOptionsFromManifest,
  getIIIFPresentationCredit,
} from '@weco/common/utils/iiif';
import NextLink from 'next/link';
import SpacingSection from '@weco/common/views/components/SpacingSection/SpacingSection';
import Icon from '@weco/common/views/components/Icon/Icon';
import CopyUrl from '@weco/common/views/components/CopyUrl/CopyUrl';
import Layout12 from '@weco/common/views/components/Layout12/Layout12';
import SpacingComponent from '@weco/common/views/components/SpacingComponent/SpacingComponent';
import Space from '@weco/common/views/components/styled/Space';
import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';
import Download from '../Download/Download';
import WorkDetailsSection from '../WorkDetailsSection/WorkDetailsSection';
import WorkDetailsText from '../WorkDetailsText/WorkDetailsText';
import WorkDetailsList from '../WorkDetailsList/WorkDetailsList';
import WorkDetailsLinks from '../WorkDetailsLinks/WorkDetailsLinks';
import WorkDetailsTags from '../WorkDetailsTags/WorkDetailsTags';
import WorkItemsStatus from '../WorkItemsStatus/WorkItemsStatus';
import type { DigitalLocation } from '@weco/common/utils/works';

type Props = {|
  work: Work,
  iiifPresentationManifest: ?IIIFManifest,
  childManifestsCount: number,
  imageCount: number,
|};

const WorkDetails = ({
  work,
  iiifPresentationManifest,
  childManifestsCount,
  imageCount,
}: Props) => {
  // Determin digital location
  const iiifImageLocation = getDigitalLocationOfType(work, 'iiif-image');
  const iiifPresentationLocation = getDigitalLocationOfType(
    work,
    'iiif-presentation'
  );
  const digitalLocation: ?DigitalLocation =
    iiifPresentationLocation || iiifImageLocation;

  // 'Available online' data
  const license =
    digitalLocation && getAugmentedLicenseInfo(digitalLocation.license);

  // iiif-presentation locations don't have credit info in the work API currently, so we try and get it from the manifest
  const credit =
    (digitalLocation && digitalLocation.credit) ||
    (iiifPresentationManifest &&
      getIIIFPresentationCredit(iiifPresentationManifest));

  const iiifImageLocationUrl = iiifImageLocation && iiifImageLocation.url;
  const iiifImageDownloadOptions = iiifImageLocationUrl
    ? getDownloadOptionsFromImageUrl(iiifImageLocationUrl)
    : [];
  const iiifPresentationDownloadOptions = iiifPresentationManifest
    ? getDownloadOptionsFromManifest(iiifPresentationManifest)
    : [];

  const downloadOptions = [
    ...iiifImageDownloadOptions,
    ...iiifPresentationDownloadOptions,
  ];

  // 'About this work' data
  const duration =
    work.duration && moment.utc(work.duration).format('HH:mm:ss');

  // 'Indentifiers' data
  const isbnIdentifiers = work.identifiers.filter(id => {
    return id.identifierType.id === 'isbn';
  });

  const sierraIdFromPresentationManifestUrl = // TODO replace with function created elsewhere
    iiifPresentationLocation &&
    (iiifPresentationLocation.url.match(/iiif\/(.*)\/manifest/) || [])[1];

  const sierraWorkIds = getWorkIdentifiersWith(work, {
    identifierId: 'sierra-system-number',
  });

  const sierraWorkId = sierraWorkIds.length >= 1 ? sierraWorkIds[0] : null;

  // We do not wish to display a link to the old library site if the new site
  // provides a digital representation of that work
  const shouldDisplayEncoreLink = !(
    sierraIdFromPresentationManifestUrl === sierraWorkId
  );
  const encoreLink =
    shouldDisplayEncoreLink && sierraWorkId && getEncoreLink(sierraWorkId);

  const locationOfWork = work.notes.find(
    note => note.noteType.id === 'location-of-original'
  );

  const WhereToFindIt = () => (
    <WorkDetailsSection headingText="Where to find it">
      {locationOfWork && (
        <WorkDetailsText
          title={locationOfWork.noteType.label}
          text={locationOfWork.contents}
        />
      )}

      {encoreLink && (
        <WorkDetailsText
          text={[`<a href="${encoreLink}">Wellcome library</a>`]}
        />
      )}

      <TogglesContext.Consumer>
        {({ stacksRequestService }) =>
          stacksRequestService && (
            <div className={`${font('hnl', 5)}`}>
              <WorkItemsStatus work={work} />
            </div>
          )
        }
      </TogglesContext.Consumer>
    </WorkDetailsSection>
  );

  return (
    <Space
      v={{
        size: 'xl',
        properties: ['padding-top', 'padding-bottom'],
      }}
      className={classNames({
        row: true,
      })}
    >
      <Layout12>
        {digitalLocation && (
          <WorkDetailsSection headingText="Available online">
            <p>
              Contains:{' '}
              {childManifestsCount > 0
                ? `${childManifestsCount} volumes`
                : imageCount > 0
                ? `${imageCount} ${imageCount === 1 ? 'image' : 'images'}`
                : ''}
            </p>

            <Download work={work} downloadOptions={downloadOptions} />
            {!(downloadOptions.length > 0) &&
              sierraIdFromPresentationManifestUrl &&
              childManifestsCount === 0 && (
                <SpacingSection>
                  <div
                    className={classNames({
                      grid: true,
                    })}
                  >
                    <div
                      className={classNames({
                        [grid({
                          s: 12,
                          m: 12,
                          l: 10,
                          xl: 10,
                        })]: true,
                      })}
                    >
                      <NextLink
                        {...downloadUrl({
                          workId: work.id,
                          sierraId: sierraIdFromPresentationManifestUrl,
                        })}
                      >
                        <a>Download options</a>
                      </NextLink>
                    </div>
                  </div>
                </SpacingSection>
              )}

            {license && (
              <div key={license.url}>
                {license.humanReadableText.length > 0 && (
                  <WorkDetailsText
                    title="License information"
                    text={license.humanReadableText}
                  />
                )}
                <WorkDetailsText
                  title="Credit"
                  text={[
                    `${work.title.replace(/\.$/g, '')}.${' '}
                ${
                  credit
                    ? `Credit: <a href="https://wellcomecollection.org/works/${work.id}">${credit}</a>. `
                    : ` `
                }
              ${
                license.url
                  ? `<a href="${license.url}">${license.label}</a>`
                  : license.label
              }`,
                  ]}
                />
              </div>
            )}
          </WorkDetailsSection>
        )}
        {!digitalLocation && (locationOfWork || encoreLink) && (
          <WhereToFindIt />
        )}
        <WorkDetailsSection headingText="About this work">
          {work.alternativeTitles.length > 0 && (
            <WorkDetailsText
              title="Also known as"
              text={work.alternativeTitles}
            />
          )}

          {work.description && (
            <WorkDetailsText title="Description" text={[work.description]} />
          )}

          {work.contributors.length > 0 && (
            <WorkDetailsTags
              title="Contributors"
              tags={work.contributors.map(contributor => ({
                textParts: [contributor.agent.label],
                linkAttributes: worksLink({
                  query: `"${contributor.agent.label}"`,
                }),
              }))}
            />
          )}

          {work.lettering && (
            <WorkDetailsText title="Lettering" text={[work.lettering]} />
          )}

          {work.production.length > 0 && (
            <WorkDetailsText
              title="Publication/Creation"
              text={work.production.map(
                productionEvent => productionEvent.label
              )}
            />
          )}

          {work.edition && (
            <WorkDetailsText title="Edition" text={[work.edition]} />
          )}

          {work.physicalDescription && (
            <WorkDetailsText
              title="Physical description"
              text={[work.physicalDescription]}
            />
          )}

          {duration && <WorkDetailsText title="Duration" text={[duration]} />}

          {work.notes
            .filter(note => note.noteType.id !== 'location-of-original')
            .map(note => (
              <WorkDetailsText
                key={note.noteType.label}
                title={note.noteType.label}
                text={note.contents}
              />
            ))}

          {work.genres.length > 0 && (
            <WorkDetailsTags
              title="Type/Technique"
              tags={work.genres.map(g => {
                return {
                  textParts: g.concepts.map(c => c.label),
                  linkAttributes: worksLink({
                    query: `"${g.label}"`,
                  }),
                };
              })}
            />
          )}

          {work.language && (
            <WorkDetailsLinks
              title="Language"
              links={[work.language && work.language.label]}
            />
          )}
        </WorkDetailsSection>
        {work.subjects.length > 0 && (
          <WorkDetailsSection headingText="Subjects">
            <WorkDetailsTags
              title={null}
              tags={work.subjects.map(s => {
                return {
                  textParts: s.concepts.map(c => c.label),
                  linkAttributes: worksLink({
                    query: `"${s.label}"`,
                  }),
                };
              })}
            />
          </WorkDetailsSection>
        )}
        {digitalLocation && (locationOfWork || encoreLink) && <WhereToFindIt />}
        <WorkDetailsSection headingText="Identifiers">
          {isbnIdentifiers.length > 0 && (
            <WorkDetailsList
              title="ISBN"
              list={isbnIdentifiers.map(id => id.value)}
            />
          )}
          <SpacingComponent>
            <div className={`${font('hnl', 5)}`}>
              <CopyUrl
                id={work.id}
                url={`https://wellcomecollection.org/works/${work.id}`}
              />
            </div>
          </SpacingComponent>
          {work.citeAs && (
            <WorkDetailsText title="Reference number" text={[work.citeAs]} />
          )}
        </WorkDetailsSection>
        <WorkDetailsSection>
          <div className="flex flex--v-center">
            <Icon name="underConstruction" extraClasses="margin-right-s2" />
            <p className={`${font('hnl', 5)} no-margin`}>
              We’re improving the information on this page.{' '}
              <a href="/works/progress">Find out more</a>.
            </p>
          </div>
        </WorkDetailsSection>
      </Layout12>
    </Space>
  );
};

export default WorkDetails;
