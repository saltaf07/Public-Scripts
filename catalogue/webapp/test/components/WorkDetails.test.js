// @flow
import {
  mountWithTheme,
  updateWrapperAsync,
} from '@weco/common/test/fixtures/enzyme-helpers';
import WorkDetails from '@weco/catalogue/components/WorkDetails/WorkDetails';
import useAuth from '@weco/common/hooks/useAuth';
import { getUserHolds } from '@weco/catalogue/services/stacks/requests';
import mockAuthStates from '@weco/catalogue/__mocks__/auth-states';
import mockWork from '@weco/catalogue/__mocks__/stacks-work';
import mockRequests from '@weco/catalogue/__mocks__/stacks-requests';
import TogglesContext from '@weco/common/views/components/TogglesContext/TogglesContext';
import 'next/router';

jest.mock('next/router', () => ({
  query: {
    code: '',
  },
}));
// const availableItem = mockWork.items.find(
//   item => item.status.label === 'Available'
// );
// const unavailableItem = mockWork.items.find(
//   item => item.status.label !== 'Available'
// );
// const unrequestableItem = mockWork.items.find(
//   item => item.status.label === 'Missing'
// );

// jest.mock('@weco/catalogue/services/stacks/requests', () => ({
//   getUserHolds: jest.fn(),
// }));
jest.mock('@weco/common/hooks/useAuth');

describe('Feature: 2. As a library member I want to request an item', () => {
  // (getUserHolds: any).mockResolvedValue(mockRequests);
  test.only(`
    Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s available to request
      Then I can see a primary CTA to log in
    `, async () => {
    (useAuth: any).mockImplementation(() => mockAuthStates.unauthorized);
    const component = mountWithTheme(
      <TogglesContext.Provider value={{ stacksRequestService: true }}>
        <WorkDetails
          work={mockWork}
          iiifPresentationManifest={null}
          itemUrl={null}
          childManifestsCount={1}
          imageCount={1}
          showAvailableOnline={true}
        />
      </TogglesContext.Provider>
    );
    await updateWrapperAsync(component);

    expect(
      component.find("[data-test-id='libraryLoginCTA']")
    ).toMatchSnapshot();
  });

  // test(`Scenario: I'm logged out, viewing a work page
  //     Given the work has a requestable item
  //     And it’s unavailable to request
  //     Then I can't see a primary CTA to log in`, async () => {
  //   (useAuth: any).mockReturnValue(mockAuthStates.unauthorized);
  //   const component = mountWithTheme(
  //     <ItemRequestButton item={unavailableItem} workId={'123'} />
  //   );
  //   await updateWrapperAsync(component);
  //   expect(component.find("[data-test-id='libraryLoginCTA']").exists()).toBe(
  //     false
  //   );
  // });

  // // we currently make no distinction between unrequestable and unavailable
  // test.skip(`Scenario: I'm logged out, viewing a work page
  //   Given the work has no requestable items
  //   Then I can't see a primary CTA to log in`, async () => {
  //   (useAuth: any).mockReturnValue(mockAuthStates.unauthorized);
  //   const component = mountWithTheme(
  //     <ItemRequestButton item={unrequestableItem} workId={'123'} />
  //   );
  //   await updateWrapperAsync(component);
  //   expect(component.find("[data-test-id='libraryLoginCTA']").exists()).toBe(
  //     false
  //   );
  // });

  // test(`Scenario: I'm logged in, viewing a work page
  // Given the work has a requestable item
  // And it’s available to request
  // Then I can see a primary CTA to request it`, async () => {
  //   (useAuth: any).mockImplementation(() => {
  //     return mockAuthStates.authorized;
  //   });
  //   const component = mountWithTheme(
  //     <ItemRequestButton item={availableItem} workId={'123'} />
  //   );
  //   await updateWrapperAsync(component);
  //   expect(component.find("[data-test-id='libraryRequestCTA']").exists()).toBe(
  //     true
  //   );
  //   expect(component.find("[data-test-id='libraryRequestCTA']").text()).toEqual(
  //     'Request to view in the library'
  //   );
  // });

  // (test: any).todo(`Scenario: A work has a single requestable item
  // Given that I am on a work page that has a single requestable item
  // When a user clicks the primary CTA to request the item
  // Then I receive feedback as to whether the request succeeded or failed
  // And if successful, I can see information about where to pick up the item`);

  // (test: any).todo(`Scenario: A work has multiple requestable items
  // Given that I am on a work page that has multiple requestable items
  // Then I am able to select which items I want to request
  // And click the primary CTA to request the items
  // Then I receive feedback as to which item request succeeded or failed
  // And if successful, I can see information about where to pick up the items`);
});
