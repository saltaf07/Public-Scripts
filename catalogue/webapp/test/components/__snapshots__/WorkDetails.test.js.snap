// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature: 2. As a library member I want to request an item 
    Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s available to request
      Then I can see a primary CTA to log in
     1`] = `
<div
  data-test-id="libraryLoginCTA"
>
  <ButtonSolidLink
    clickHandler={[Function]}
    link="/loginUrl"
    text="Log in to request"
    trackingEvent={
      Object {
        "action": "follow Cognito login link",
        "category": "Button",
        "label": "{workId}",
      }
    }
  >
    <ConditionalWrapper
      condition={false}
      wrapper={[Function]}
    >
      <ButtonSolid__SolidButton
        href="/loginUrl"
        onClick={[Function]}
      >
        <a
          className="ButtonSolid__BaseButton-is2j2c-0 ButtonSolid__SolidButton-is2j2c-4 gGrRBp dCZQTx link-reset"
          href="/loginUrl"
          onClick={[Function]}
        >
          <ButtonSolid__BaseButtonInner>
            <span
              className="ButtonSolid__BaseButtonInnerSpan-is2j2c-1 ButtonSolid__BaseButtonInner-is2j2c-2  jMNuud font-hnm font-size-5  flex flex--v-center"
            >
              Log in to request
            </span>
          </ButtonSolid__BaseButtonInner>
        </a>
      </ButtonSolid__SolidButton>
    </ConditionalWrapper>
  </ButtonSolidLink>
</div>
`;

exports[`Feature: 2. As a library member I want to request an item Scenario: I'm logged in, viewing a work page
  Given the work has a requestable item
  And it’s available to request
  Then I can see a primary CTA to request it 1`] = `
<div
  data-test-id="requestModalCTA"
>
  <ButtonSolid
    clickHandler={[Function]}
    text="Request"
    trackingEvent={
      Object {
        "action": "open Stacks request modal window",
        "category": "Button",
        "label": "{workId}",
      }
    }
  >
    <ButtonSolid__SolidButton
      onClick={[Function]}
    >
      <button
        className="ButtonSolid__BaseButton-is2j2c-0 ButtonSolid__SolidButton-is2j2c-4 gGrRBp dCZQTx"
        onClick={[Function]}
      >
        <ButtonSolid__BaseButtonInner>
          <span
            className="ButtonSolid__BaseButtonInnerSpan-is2j2c-1 ButtonSolid__BaseButtonInner-is2j2c-2  jMNuud font-hnm font-size-5  flex flex--v-center"
          >
            <span
              className=""
            >
              Request
            </span>
          </span>
        </ButtonSolid__BaseButtonInner>
      </button>
    </ButtonSolid__SolidButton>
  </ButtonSolid>
  <Modal
    id="requestModal"
    isActive={false}
    openButtonRef={
      Object {
        "current": <button
          class="ButtonSolid__BaseButton-is2j2c-0 ButtonSolid__SolidButton-is2j2c-4 gGrRBp dCZQTx"
        >
          <span
            class="ButtonSolid__BaseButtonInnerSpan-is2j2c-1 ButtonSolid__BaseButtonInner-is2j2c-2  jMNuud font-hnm font-size-5  flex flex--v-center"
          >
            <span
              class=""
            >
              Request
            </span>
          </span>
        </button>,
      }
    }
    setIsActive={[Function]}
  >
    <CSSTransition
      classNames="fade"
      in={false}
      timeout={350}
    >
      <Transition
        appear={false}
        enter={true}
        exit={true}
        in={false}
        mountOnEnter={false}
        onEnter={[Function]}
        onEntered={[Function]}
        onEntering={[Function]}
        onExit={[Function]}
        onExited={[Function]}
        onExiting={[Function]}
        timeout={350}
        unmountOnExit={false}
      >
        <Modal__ModalWindow
          hidden={true}
          id="requestModal"
          width={null}
        >
          <div
            className="Space-sc-4az8w3-0 Modal__ModalWindow-sc-12234yr-2 qQBZz deVjSn shadow bg-white"
            hidden={true}
            id="requestModal"
            width={null}
          >
            <Modal__CloseButton
              hideFocus={false}
              onClick={[Function]}
            >
              <button
                className="Space-sc-4az8w3-0 Modal__CloseButton-sc-12234yr-1 bkjmsI hctMRG"
                onClick={[Function]}
                role="button"
              >
                <span
                  className="visually-hidden"
                >
                  Close modal window
                </span>
                <Icon
                  extraClasses="icon--currentColor"
                  name="cross"
                >
                  <div
                    className="icon icon--currentColor"
                  >
                    <svg
                      aria-hidden={true}
                      className="icon__svg"
                    >
                      <svg
                        viewBox="0 0 24 24"
                      >
                        <path
                          className="icon__shape"
                          d="M18.7 17.3L13.4 12l5.3-5.3c.4-.4.4-1 0-1.4-.4-.4-1-.4-1.4 0L12 10.6 6.7 5.3c-.4-.4-1-.4-1.4 0-.4.4-.4 1 0 1.4l5.3 5.3-5.3 5.3c-.4.4-.4 1 0 1.4.4.4 1 .4 1.4 0l5.3-5.3 5.3 5.3c.4.4 1 .4 1.4 0 .4-.4.4-1 0-1.4z"
                          fillRule="nonzero"
                        />
                      </svg>
                    </svg>
                  </div>
                </Icon>
              </button>
            </Modal__CloseButton>
            <h2
              className="font-hnm font-size-5 "
            >
              Request items
            </h2>
            <p
              className="font-hnl font-size-5 "
            >
              You are about to request the following items:
            </p>
            <p
              className="font-hnm font-size-6 "
            >
              Sir Joseph Banks : Royal Botanical Garden St. Vincent 1764-1820.
            </p>
            <ul
              className="plain-list no-padding"
            >
              <li
                className="font-hnm font-size-5 "
                key="{itemIdAvailable}"
              >
                <Space
                  as="span"
                  className="font-hnl font-size-5 "
                  h={
                    Object {
                      "properties": Array [
                        "margin-right",
                      ],
                      "size": "s",
                    }
                  }
                >
                  <span
                    className="Space-sc-4az8w3-0 eFqFvD font-hnl font-size-5 "
                  >
                    <label
                      className="visually-hidden"
                    >
                      Request 
                      {itemIdAvailable}
                    </label>
                    <CheckboxRadio
                      checked={true}
                      id="{itemIdAvailable}"
                      name="{itemIdAvailable}"
                      onChange={[Function]}
                      text=""
                      type="checkbox"
                      value="{itemIdAvailable}"
                    >
                      <CheckboxRadio__CheckboxRadioLabel
                        htmlFor="{itemIdAvailable}"
                      >
                        <label
                          className="CheckboxRadio__CheckboxRadioLabel-sc-18nuwsd-0 eiIodU flex-inline flex--v-center"
                          htmlFor="{itemIdAvailable}"
                        >
                          <CheckboxRadio__CheckboxRadioInput
                            checked={true}
                            id="{itemIdAvailable}"
                            name="{itemIdAvailable}"
                            onChange={[Function]}
                            type="checkbox"
                            value="{itemIdAvailable}"
                          >
                            <input
                              checked={true}
                              className="CheckboxRadio__CheckboxRadioInput-sc-18nuwsd-3 bvVdNq"
                              id="{itemIdAvailable}"
                              name="{itemIdAvailable}"
                              onChange={[Function]}
                              type="checkbox"
                              value="{itemIdAvailable}"
                            />
                          </CheckboxRadio__CheckboxRadioInput>
                          <CheckboxRadio__CheckboxRadioBox
                            type="checkbox"
                          >
                            <span
                              className="CheckboxRadio__CheckboxRadioBoxSpan-sc-18nuwsd-1 CheckboxRadio__CheckboxRadioBox-sc-18nuwsd-2  uVpdx flex-inline flex--v-center flex--h-center relative"
                              type="checkbox"
                            >
                              <Icon
                                name="check"
                              >
                                <div
                                  className="icon "
                                >
                                  <svg
                                    aria-hidden={true}
                                    className="icon__svg"
                                  >
                                    <svg
                                      viewBox="0 0 24 24"
                                    >
                                      <path
                                        className="icon__shape"
                                        d="M21.13 6.64a1 1 0 0 0-1.41 0L10 16.34l-5-5a1 1 0 0 0-1.41 1.41l5.67 5.67a1 1 0 0 0 1.41 0l10.4-10.4a1 1 0 0 0 .06-1.38z"
                                        fillRule="nonzero"
                                      />
                                    </svg>
                                  </svg>
                                </div>
                              </Icon>
                            </span>
                          </CheckboxRadio__CheckboxRadioBox>
                          <Space
                            as="span"
                            h={
                              Object {
                                "properties": Array [
                                  "margin-left",
                                ],
                                "size": "xs",
                              }
                            }
                          >
                            <span
                              className="Space-sc-4az8w3-0 jPPhdA"
                            />
                          </Space>
                        </label>
                      </CheckboxRadio__CheckboxRadioLabel>
                    </CheckboxRadio>
                  </span>
                </Space>
                <Space
                  as="span"
                  className="font-hnl font-size-5 "
                  h={
                    Object {
                      "properties": Array [
                        "margin-left",
                      ],
                      "size": "l",
                    }
                  }
                >
                  <span
                    className="Space-sc-4az8w3-0 kmFYVg font-hnl font-size-5 "
                  >
                    Closed stores Biog. pam
                  </span>
                </Space>
              </li>
            </ul>
            <div>
              <ItemRequestButton
                itemsWithPhysicalLocations={
                  Array [
                    Object {
                      "checked": true,
                      "dueDate": "{dueDate}",
                      "id": "{itemIdAvailable}",
                      "identifiers": Array [
                        Object {
                          "identifierType": Object {
                            "id": "sierra-system-number",
                            "label": "Sierra system number",
                            "type": "IdentifierType",
                          },
                          "type": "Identifier",
                          "value": "i10187431",
                        },
                        Object {
                          "identifierType": Object {
                            "id": "sierra-identifier",
                            "label": "Sierra identifier",
                            "type": "IdentifierType",
                          },
                          "type": "Identifier",
                          "value": "1018743",
                        },
                      ],
                      "locations": Array [
                        Object {
                          "accessConditions": Array [],
                          "label": "Closed stores Biog. pam",
                          "locationType": Object {
                            "id": "sgpbi",
                            "label": "Closed stores Biog. pam",
                            "type": "LocationType",
                          },
                          "type": "PhysicalLocation",
                        },
                      ],
                      "requestable": true,
                      "status": Object {
                        "id": "-",
                        "label": "Available",
                        "type": "ItemStatus",
                      },
                      "type": "Item",
                    },
                  ]
                }
                setItemsWithPhysicalLocations={[Function]}
                setShowRequestModal={[Function]}
                setShowResultsModal={[Function]}
              >
                <ButtonSolid
                  clickHandler={[Function]}
                  text="Request to view in library"
                  trackingEvent={
                    Object {
                      "action": "confirm Stacks request",
                      "category": "Button",
                      "label": "item id(s): {itemIdAvailable}",
                    }
                  }
                >
                  <ButtonSolid__SolidButton
                    onClick={[Function]}
                  >
                    <button
                      className="ButtonSolid__BaseButton-is2j2c-0 ButtonSolid__SolidButton-is2j2c-4 gGrRBp dCZQTx"
                      onClick={[Function]}
                    >
                      <ButtonSolid__BaseButtonInner>
                        <span
                          className="ButtonSolid__BaseButtonInnerSpan-is2j2c-1 ButtonSolid__BaseButtonInner-is2j2c-2  jMNuud font-hnm font-size-5  flex flex--v-center"
                        >
                          <span
                            className=""
                          >
                            Request to view in library
                          </span>
                        </span>
                      </ButtonSolid__BaseButtonInner>
                    </button>
                  </ButtonSolid__SolidButton>
                </ButtonSolid>
              </ItemRequestButton>
              <button
                className="plain-button"
                onClick={[Function]}
              >
                <Space
                  as="span"
                  className="font-hnl font-size-6 "
                  h={
                    Object {
                      "properties": Array [
                        "margin-left",
                      ],
                      "size": "l",
                    }
                  }
                >
                  <span
                    className="Space-sc-4az8w3-0 kmFYVg font-hnl font-size-6 "
                  >
                    Cancel
                  </span>
                </Space>
              </button>
            </div>
          </div>
        </Modal__ModalWindow>
      </Transition>
    </CSSTransition>
  </Modal>
</div>
`;

exports[`Feature: 2. As a library member I want to request an item Scenario: I'm logged out, viewing a work page
      Given the work has a requestable item
      And it’s unavailable to request
      Then I can't see a primary CTA to log in 1`] = `null`;
